lib/rucio/daemons/rsedecommissioner/profiles/atlas.py

```diff
@@ -0,0 +1,76 @@
+# -*- coding: utf-8 -*-
+# Copyright European Organization for Nuclear Research (CERN) since 2012
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""ATLAS-specific decommissioning profiles."""
+
+import logging
+from collections.abc import Callable
+from typing import TYPE_CHECKING, Any
+
+from rucio.core.did import get_metadata, list_content
+from rucio.core.replica import declare_bad_file_replicas
+
+from .generic import generic_move
+
+if TYPE_CHECKING:
+    from ..profile import DecommissioningProfile
+
+
+def atlas_move(rse: dict[str, Any], config: dict[str, Any]) -> "DecommissioningProfile":
+    """Return a profile for moving rules that satisfy conditions to a specific destination.
+
+    The "ATLAS move" profile lists out all rules that are locking replicas
+    at the given RSE, and moves them to the specified destination if either
+    one of the following is true:
+
+    - The RSE expression of the rule is trivial (the RSE name itself).
+    - There are no replicas locked by the rule that reside on another RSE.
+    - The datatype of the DID is not "log".
+
+    :param rse: RSE to decommission.
+    :param config: Decommissioning configuration dictionary.
+    :returns: A decommissioning profile dictionary.
+    """
+    profile = generic_move(rse, config)
+    # Insert before the trivial RSE expression handler
+    idx = next(pos for pos, handler in enumerate(profile.handlers)
+               if handler[0].__name__ == '_has_trivial_rse_expression')
+    profile.handlers.insert(idx, (_is_log_file, _declare_files_as_bad))
+    return profile
+
+
+def _is_log_file(
+    rule: dict[str, Any],
+    rse: dict[str, Any],
+    *,
+    logger: Callable[..., None] = logging.log
+) -> bool:
+    """Check if the datatype metadata is 'log'."""
+    return get_metadata(rule['scope'], rule['name'])['datatype'] == 'log'
+
+
+def _declare_files_as_bad(
+    rule: dict[str, Any],
+    rse: dict[str, Any],
+    *,
+    logger: Callable[..., None] = logging.log
+) -> bool:
+    """Invalidate the files to detach them from a dataset."""
+    replicas = [{'scope': d['scope'], 'name': d['name'], 'rse_id': rse['id']}
+                for d in list_content(rule['scope'], rule['name'])]
+    declare_bad_file_replicas(replicas, reason='decommissioning', issuer='rsedecommissioner',
+                              force=True)
```

Comment:
I’m afraid this is dangerous. If the replicas are not unique, then Rucio will initiate a recovery from other sources. Please decide whether to implement this properly or to remove it (and replace it with a function that only logs a warning, asking for manual intervention).

Reply:
I'm not sure what the proper implementation should be at this moment, so I will change this to just call for manual intervention (i.e., return NEED_ATTENTION).

Reply:
Here’s a simplification of what DDM Ops do:

1. Declare unique replicas as bad.
2. Handle non-unique replicas at other RSEs
3. Repeat steps 1 and 2 until all logs are removed.

You don’t need to implement this. It is paramount to have a clear ATLAS-specific example, but I will justify the existing implementation as ‘policy needs to be defined and implemented by DDM Ops’.